lib/promotion.rb

```diff
@@ -26,7 +26,9 @@ def review
 
   def review_tl0
     if Promotion.tl1_met?(@user) && change_trust_level!(TrustLevel[1])
-      @user.enqueue_member_welcome_message unless @user.badges.where(id: Badge::BasicUser).count > 0
+      if Badge.find(Badge::BasicUser).enabled? && !@user.badges.where(id: Badge::BasicUser).exists?
```

Comment:
```suggestion
      if Badge.exists?(id: Badge::BasicUser, enabled: true) && !@user.badges.exists?(id: Badge::BasicUser)
```

Reply:
This will execute two queries but if we really want to optimise this, we can consider doing an inner join.

Reply:
@tgxworld what do you think about 2 selects combined? I updated PR